metatensor-core/src/io/load.rs

```diff
@@ -1,15 +1,45 @@
 use std::collections::HashSet;
+use std::io::BufReader;
 use std::sync::Arc;
 
 use byteorder::{LittleEndian, BigEndian, ReadBytesExt};
 use zip::ZipArchive;
 
 use crate::{TensorMap, TensorBlock, Labels, Error, mts_array_t};
 
-use super::check_for_extra_bytes;
-use super::labels::read_npy_labels;
+use super::{check_for_extra_bytes, PathOrBuffer};
+use super::labels::load_labels;
 use super::npy_header::{Header, DataType};
 
+pub fn looks_like_tensormap_data(mut data: PathOrBuffer) -> bool {
```

Comment:
Maybe small doc, not sure if I understand correctly so here as suggestion
```suggestion
/// Extracts from the path or buffer a ZipArchive and considers it to be a TensorMap if this archive contains a "keys.npy" entry.
pub fn looks_like_tensormap_data(mut data: PathOrBuffer) -> bool {
```

Reply:
done!